pennylane/transforms/get_unitary_matrix.py

```diff
@@ -0,0 +1,119 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A transform to obtain the matrix representation of a quantum circuit.
+"""
+from functools import wraps
+import numpy as np
+from pennylane.wires import Wires
+import pennylane as qml
+
+
+def get_unitary_matrix(circuit, wire_order=None):
+    """Construct the matrix representation of a quantum circuit.
+
+    Args:
+        circuit (.QNode, .QuantumTape, or Callable): A quantum node, tape, or function that applies quantum operations.
+        wire_order (Sequence[Any]): Order of the wires in the quantum circuit. Optional if ``circuit`` is a QNode or tape.
+
+    Returns:
+         function: Function which accepts the same arguments as the QNode or quantum function.
+         When called, this function will return the unitary matrix.
+
+    **Example**
+
+    Consider the following function:
+
+    .. code-block:: python3
+
+        def circuit(theta):
+            qml.RX(theta, wires=0)
+            qml.PauliZ(wires=1)
+
+    Choosing a wire order, we can use ``get_unitary_matrix`` to generate a new function
+    that returns the unitary matrix corresponding to the function ``circuit``:
+
+    >>> wires = [0, 1]
+    >>> get_matrix = get_unitary_matrix(circuit, wires)
+    >>> theta = np.pi/4
+    >>> get_matrix(theta)
+    array([[ 0.92387953+0.j,    0.+0.j,    0.-0.38268343j,    0.+0.j],
+    [0.+0.j,    -0.92387953+0.j,    0.+0.j,    0.+0.38268343j],
+    [0.-0.38268343j,    0.+0.j,    0.92387953+0.j,    0.+0.j],
+    [0.+0.j,    0.+0.38268343j,    0.+0.j,    -0.92387953+0.j]])
+    """
+
+    wires = wire_order
+
+    @wraps(circuit)
+    def wrapper(*args, **kwargs):
+
+        if isinstance(circuit, qml.QNode):
+            # user passed a QNode, get the tape
+            circuit.construct(args, kwargs)
+            tape = circuit.qtape
+            if wires is None:  # if no wire ordering is specified, take wire list from tape
+                wire_order = tape.wires
+            else:
+                wire_order = Wires(wires)
+
+        elif isinstance(circuit, qml.tape.QuantumTape):
+            # user passed a tape
+            tape = circuit
+            if wires is None:
+                wire_order = tape.wires
+            else:
+                wire_order = Wires(wires)
+
+        elif callable(circuit):
+            # user passed something that is callable but not a tape or qnode.
+            tape = qml.transforms.make_tape(circuit)(*args, **kwargs)
+            # raise exception if it is not a quantum function
+            if len(tape.operations) == 0:
+                raise ValueError("Function contains no quantum operation")
+            if wires is None:
```

Comment:
Why don't we use `tape.wires` here?

Reply:
I did not realize that was an option :) Will do that now.

Reply:
Yes I agree that it can be misleading. Personally I don't think we should make the `wire_order` as a required argument, but we should make clear in the documentation how is the default `wire_order` built. One thing you should also add is a check that the `wire_order` argument contains the same wires as the different tapes. At the moment if you can give wires that are not in the circuit and it will not return a specific and helpful message.